CMS-Enterprise/sbom-harbor
# Refactor sdk

## Note to reviewers

This PR consolidates existing directories that together constitute the Harbor SDK under a common parent folder named `sdk`.  It is safe for reviewers to ignore the file move changes.

## Summary

This PR introduces a Proof of Concept for adding extensions to Harbor. It uses the Snyk OrgTags/FISMA ID solution as an initial test case.

### Added

- Extension Template & Example (`sdk/extension-template`)
- CMS Custom Extension CLI project (`extensions/cms`)
- CMS `fisma` command to handle syncing FISMA IDs from the Snyk API
- New `TaskKind::Extension` enum value

### Changed

Moved all SDK created folders under a top level `sdk` folder including:

- `devenv` folder
- `generators`
- `openapi`

## How to test

- `cargo build`
- Optionally, to preserve any test data you have, move your data directory from the old `devenv/mongodb/data` folder to 
  `sdk/devenv/mongodb/data`.
- Start the devenv from the new `sdk/devenv` directory
- If you have not ingested or enriched sboms using the snyk provider, do so
- Run `./target/debug/harbor-cms fisma --debug`
- Verify that all `Package` entities of `kind: Primary` have a `fisma` Xref.
